draft-ietf-mls-protocol.md

```diff
@@ -3051,10 +3068,11 @@ External Commits work like regular Commits, with a few differences:
 * The proposals included by value in an External Commit MUST meet the following
   conditions:
   * There MUST be a single ExternalInit proposal
-  * There MUST NOT be any Add or Update proposals
-  * If a Remove proposal is present, then the `credential` and `endpoint_id` of
-    the removed leaf MUST be the same as the corresponding values in the Add
-    KeyPackage.
+  * There MUST NOT be any Update proposals
+  * If a Remove proposal is present, then the `credential` of the Add KeyPackage
+    MUST present a set of identifiers that is acceptable to the application for
+    the removed participant (as if this were an Update for that participant).
+    The Add KeyPackage and the removed leaf MUST have the same `endpoint_id`.
```

Comment:
In other words, this gives us a new requirement for the AS, namely that it has to be able to tell us if a given credential is the valid successor of another credential. With fixed identities, this would not be necessary. Of course one way of implementing this check for the AS is to check if some notion of identity that arises from a given credential type is indeed the same for both certificates.

Reply:
I wouldn't frame this as an AS requirement, but regardless, you have to do *something* here: The application has to either (1) verify that the "identity" is the same (for some definition thereof), or (2) verify that the identity is acceptable. 

(2) is the state of the art basically everywhere right now.  See, e.g., [RFC 6125](https://datatracker.ietf.org/doc/html/rfc6125).  Doing (1) for an X.509 certificate is actually pretty complex -- or even defining it -- given all the notions of "identity" lurking around in there.  For example, we could enumerate the subjectAltName, but then someone defines [a new way of embedding telephone numbers](https://datatracker.ietf.org/doc/html/rfc8226).  The only way to be sure is to require that the whole certificate be the same, but you probably also want to allow signature key rotation, so really you want the whole cert to be the same except for the public key.  This is not functionality provided by any X.509 library that I'm aware of.

I think the underlying confusion here is around what layers are responsible for what.  Compare to HTTPS:

* The application layer provides the identifier you're supposed to be authenticating (host name in the URL)
* TLS only provides a *set of identifiers* that are authenticated
* It's up the the application layer to verify that the authenticated identifiers are acceptable

(This is true even with extensions like [exported authenticators](https://datatracker.ietf.org/doc/draft-ietf-tls-exported-authenticator/) that allow the set of identifiers to change.) All this PR is proposing is a similar division of labor in the MLS case.